### PR TITLE
Allow adding new hosts to a running noping session

### PR DESCRIPTION
**Pull request by hamishcoleman**

### CODE REVIEW

**Comment by benhylau** [comment]
This would be extremely useful

File: src/oping.c [hunk 1]
```diff
@@ -1046,6 +1068,10 @@ static int update_graph_prettyping (ping_context_t *ctx, /* {{{ */
 		index = (history_offset + x) % ctx->history_size;
 		latency = ctx->history_by_time[index];
 
+                if (latency < 0) {
```
Comment (octo) [inline src/oping.c]: I don't think this is correct: there is an explicit case handling `latency < 0` down below. This happens when no response is received, I think.

File: src/oping.c [hunk 2]
```diff
@@ -1332,6 +1352,31 @@ static int check_resize (pingobj_t *ping) /* {{{ */
 			else if (opt_show_graph > 0)
 				opt_show_graph++;
 		}
+                else if (key == 'a')
+                {
+                        char host[80];
```
Comment (octo) [inline src/oping.c]: Please use `NI_MAXHOST` instead of the hard-coded 80 literal.

File: src/oping.c [hunk 3]
```diff
@@ -1332,6 +1352,31 @@ static int check_resize (pingobj_t *ping) /* {{{ */
...
+
+				wprintw (main_win, "Adding host `%s' failed: %s\n", host, errmsg);
+			}
+			else
+			{
+                                /* FIXME - scroll main_win correctly so that
+                                    old data is still visible */
+
+                                need_resize = 1;
+				host_num++;
```
Comment (octo) [inline src/oping.c]: Please unify whitespace handling. Make sure all added lines used tabs for indentation.

File: src/oping.c (hunk 1 above)
Comment (hamishcoleman) [inline src/oping.c]: liboping documented "less than zero" as no response received and uses a hard-coded -1.0, oping.c then converts any <0 latency into a NAN for storage in the history_by_time.  This looks like it made it simpler to calculate the averages and histograms than just using the raw -1 value.

So, there is some inconsistency, but it all works (eg: new hosts get space chars for new data, missed replys get red bangs)

**Comment by octo** [comment]
Thanks again @hamishcoleman!

**Pull request merged**